### ISSUE TITLE
Abstraction based Command System

### ISSUE THREAD

**Issue by Proximyst** [Proximyst/SpigotVult#24]
Create a command system which is independent on plugin.yml, kind of like this one: https://gitlab.com/proximyst/HellishSurvival/blob/master/Plugin/src/main/java/ml/proximyst/hellishsurvival/commands/CustomCommand.java

**Comment by ysl3000** [comment]
Or that. https://github.com/ysl3000/CommandAPI

**Comment by ysl3000** [comment]
https://github.com/ysl3000/CommandAPI
Needs some rework. Written 2013 like.
Dependency injection wasn't planned well.
At this time I had a crush on singleton.

**Comment by MiniDigger** [comment]
a different approach would be using annotations: https://github.com/minnymin3/CommandFramework
I think the key thing tho is making the registration cleaner. injecting into the command map is ugly as hell, we could just expose the method bukkit uses to register plugin commands that are defined in the plugin.yml

**Comment by ysl3000** [comment]
@MiniDigger i think the framework you posted is like the sk89q command framework or has similarities.  https://github.com/OvercastNetwork/sk89q-command-framework

**Comment by MiniDigger** [comment]
yeah. I think there are many similar. I got my own implementation too.
But speaking of sk89q, we could also take a look at intake :P
https://github.com/sk89q/Intake

**Comment by ysl3000** [comment]
@MiniDigger I think the commandframework was a preproject of intake.
